layers/+lang/csharp/funcs.el

```diff
@@ -0,0 +1,150 @@
+;;; funcs.el --- C# Layer functions File for Spacemacs
+;;
+;; Copyright (c) 2012-2019 Sylvain Benner & Contributors
+;;
+;; Author: Muneeb Shaikh <muneeb@reversehack.in>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+
+;; backend
+
+(defun spacemacs//csharp-backend ()
+  "Returns selected backend."
+  (if csharp-backend
+      csharp-backend
+    (cond
+     ((configuration-layer/layer-used-p 'lsp) 'lsp)
+     (t 'omnisharp))))
+
+(defun spacemacs//csharp-setup-backend ()
+  "Conditionally setup layer csharp based on backend."
+  (pcase (spacemacs//csharp-backend)
+    (`omnisharp (spacemacs//csharp-setup-omnisharp))
+    (`lsp (spacemacs//csharp-setup-lsp))))
+
+(defun spacemacs//csharp-setup-company ()
+  "Conditionally setup company based on backend."
+  (pcase (spacemacs//csharp-backend)
+    (`omnisharp (spacemacs//csharp-setup-omnisharp-company))
+    (`lsp (spacemacs//csharp-setup-lsp-company))))
+
+(defun spacemacs//csharp-configure ()
+  "Conditionally configure csharp layer based on backend."
+  (pcase (spacemacs//csharp-backend)
+    (`omnisharp (spacemacs//csharp-configure-omnisharp))
+    (`lsp (spacemacs//csharp-configure-lsp))))
```

Comment:
Please remove this line, this call is unnecessary

Reply:
ok